File: DetailViewTabCount/Module/VB/DetailViewTabCountVB.Module/DatabaseUpdate/Updater.vb
Option Infer On
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Persistent.BaseImpl

Public Class Updater
    Inherits ModuleUpdater

    Public Sub New(ByVal objectSpace As IObjectSpace, ByVal currentDBVersion As Version)
        MyBase.New(objectSpace, currentDBVersion)
    End Sub
    Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
        MyBase.UpdateDatabaseAfterUpdateSchema()

        Dim employee = ObjectSpace.FindObject(Of Employee)(Nothing)
        If employee Is Nothing Then
            Dim maria = ObjectSpace.CreateObject(Of Employee)()
            maria.FirstName = "Maria"
            maria.LastName = "Anders"

            Dim ana = ObjectSpace.CreateObject(Of Employee)()
            ana.FirstName = "Ana"
            ana.LastName = "Trujillo"

            Dim antonio = ObjectSpace.CreateObject(Of Employee)()
            antonio.FirstName = "Antonio"
            antonio.LastName = "Moreno"

            Dim task = ObjectSpace.CreateObject(Of Task)()
            task.Subject = "Ana's task 1"
            task.AssignedTo = ana
            task = ObjectSpace.CreateObject(Of Task)()
            task.Subject = "Ana's task 2"
            task.AssignedTo = ana

            task = ObjectSpace.CreateObject(Of Task)()
            task.Subject = "Antonio's task"
            task.AssignedTo = antonio

            Dim phoneNumber = ObjectSpace.CreateObject(Of PhoneNumber)()
            phoneNumber.Number = "+1234567890"
            phoneNumber.Party = ana

            phoneNumber = ObjectSpace.CreateObject(Of PhoneNumber)()
            phoneNumber.Number = "+1234567890"
            phoneNumber.Party = antonio
        End If

        ObjectSpace.CommitChanges() 'Uncomment this line to persist created object(s).
    End Sub
    Public Overrides Sub UpdateDatabaseBeforeUpdateSchema()
        MyBase.UpdateDatabaseBeforeUpdateSchema()
        'if(CurrentDBVersion < new Version("1.1.0.0") && CurrentDBVersion > new Version("0.0.0.0")) {
        '    RenameColumn("DomainObject1Table", "OldColumnName", "NewColumnName");
        '}
    End Sub
End Class

File: DetailViewTabCount/Module/CS/DetailViewTabCount.Module/DatabaseUpdate/Updater.cs
using System;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Updating;
using DetailViewTabCount.Module.BusinessObjects;
using DevExpress.Persistent.BaseImpl;
using Task = DetailViewTabCount.Module.BusinessObjects.Task;

namespace DetailViewTabCount.Module.DatabaseUpdate
{
    // For more typical usage scenarios, be sure to check out https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.Updating.ModuleUpdater
    public class Updater : ModuleUpdater {
        public Updater(IObjectSpace objectSpace, Version currentDBVersion) :
            base(objectSpace, currentDBVersion) {
        }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();

            var employee = ObjectSpace.FindObject<Employee>(null);
            if (employee == null)
            {
                var maria = ObjectSpace.CreateObject<Employee>();
                maria.FirstName = "Maria";
                maria.LastName = "Anders";

                var ana = ObjectSpace.CreateObject<Employee>();
                ana.FirstName = "Ana";
                ana.LastName = "Trujillo";

                var antonio = ObjectSpace.CreateObject<Employee>();
                antonio.FirstName = "Antonio";
                antonio.LastName = "Moreno";

                var task = ObjectSpace.CreateObject<Task>();
                task.Subject = "Ana's task 1";
                task.AssignedTo = ana;
                task = ObjectSpace.CreateObject<Task>();
                task.Subject = "Ana's task 2";
                task.AssignedTo = ana;

                task = ObjectSpace.CreateObject<Task>();
                task.Subject = "Antonio's task";
                task.AssignedTo = antonio;

                var phoneNumber = ObjectSpace.CreateObject<PhoneNumber>();
                phoneNumber.Number = "+1234567890";
                phoneNumber.Party = ana;

                phoneNumber = ObjectSpace.CreateObject<PhoneNumber>();
                phoneNumber.Number = "+1234567890";
                phoneNumber.Party = antonio;
            }

            ObjectSpace.CommitChanges(); //Uncomment this line to persist created object(s).
        }
        public override void UpdateDatabaseBeforeUpdateSchema() {
            base.UpdateDatabaseBeforeUpdateSchema();
            //if(CurrentDBVersion < new Version("1.1.0.0") && CurrentDBVersion > new Version("0.0.0.0")) {
            //    RenameColumn("DomainObject1Table", "OldColumnName", "NewColumnName");
            //}
        }
    }
}
